### ISSUE TITLE
balance --budget should respect -H/--historical, --cumulative

### ISSUE THREAD

**Issue by simonmichael** [simonmichael/hledger#1879]
Reported by dentalfloss on chat: balance --budget respects -H/--historical and --cumulative for actual amounts but not for goal amounts. We'd like to fix this, I am looking into it.

**Comment by simonmichael** [comment]
A new failing example from dentalfloss. The date span within the periodic rule should also be enforced:
```
~ every february 2nd from 2020/01
    Expenses:Operating:CME      414.00 CAD
    Assets:Chequing
    
#  hledger -fbudget.acct balance --budget -MH -b2020/01 -e2020/04
# Budget performance in 2020Q1:
# 
#                         ||            2020-01-31             2020-02-29             2020-03-31
# ========================++=====================================================================
#  Assets:Chequing        || 0 [0% of -414.00 CAD]  0 [0% of -828.00 CAD]  0 [0% of -828.00 CAD]
#  Expenses:Operating:CME || 0 [0% of  414.00 CAD]  0 [0% of  828.00 CAD]  0 [0% of  828.00 CAD]
# ------------------------++---------------------------------------------------------------------
#                         || 0 [                0]  0 [                0]  0 [                0]
                        
# hledger -fbudget.acct balance --budget -DH -b2020/02/01 -e2020/02/07
# Budget performance in 2020-02-01..2020-02-06:
# 
#                         ||            2020-02-01             2020-02-02             2020-02-03             2020-02-04             2020-02-05             2020-02-06
# ========================++==========================================================================================================================================
#  Assets:Chequing        || 0 [0% of -414.00 CAD]  0 [0% of -828.00 CAD]  0 [0% of -828.00 CAD]  0 [0% of -828.00 CAD]  0 [0% of -828.00 CAD]  0 [0% of -828.00 CAD]
#  Expenses:Operating:CME || 0 [0% of  414.00 CAD]  0 [0% of  828.00 CAD]  0 [0% of  828.00 CAD]  0 [0% of  828.00 CAD]  0 [0% of  828.00 CAD]  0 [0% of  828.00 CAD]
# ------------------------++------------------------------------------------------------------------------------------------------------------------------------------
#                         || 0 [                0]  0 [                0]  0 [                0]  0 [                0]  0 [                0]  0 [                0]
```

**Comment by simonmichael** [comment]
Fixed by 7068517f2.
